doc/sonic-build-system/web_file_server_population_script.md

```diff
@@ -0,0 +1,129 @@
+
+| Name        	| Description                                                                                                	| Tag          	| Type    	| Default value                      	|
+|-------------	|------------------------------------------------------------------------------------------------------------	|--------------	|---------	|------------------------------------	|
+| source      	| Search path for source tree where to  look for<br> compilation artifacts (version files)                   	| -s, --source 	| String  	| CWD<br>(current working directory) 	|
+| cache       	| Path for temporary storage where to download Web Packages                                                  	| -c, --cache  	| String  	| CWD/tmp                            	|
+| output      	| Output file name to hold the list of packages.<br> The file will be created in the format of version files 	| -o, --output 	| String  	| none                               	|
+| print       	| Print level verbosity                                                                                      	| -p, --print  	| integer 	| 1 (info level)                     	|
+| user        	| User for trusted server authentication                                                                     	| -u, --user   	| String  	| none                               	|
+| key         	| key server authentication                                                                                  	| -k, --key    	| String  	| none                               	|
+| Destination 	| URL for destination web file server                                                                        	| -d, --dest   	| String  	| none                               	|
+
+
+## Script outputs:
+
+The script would print out to STDOUT its current download progress (progress bar) and error messages in case of errors.
+The script can produce a list of packages that were uploaded in a format of version file.
+The script would return 0 in case of success and an errors code (see table below) in case of failure.
+### Error codes:
+
+Code         | Description					        | Comments
+------------ | -------------------------------------| -------------
+ 0		     | Success						        |
+-1		     | Cannot write to cache                |
+-2		     | Cannot access trusted server         |
+-3		     | Cannot authenticate trusted server   |
+
+
+## Build flow integration:
+Script will be integrated into SONiC build environment as part of “make freeze” where invocation of the script is optional. User may choose to call script by defining following environment variables:
+
+```sh
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_ENABLE=1
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_USER=<User for trusted server authentication>
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_KEY=<API key server authentication>
+export SONIC_REPRODUCABLE_BUILD_UPLAOD_DEST=< URL for destination web file server>
+```
```

Comment:
Can we download the packages to a specified folder? Un a build, we only care what kinds of artifacts it will generate, but how to publish it might be another story.
The Different web servers have different ways to upload files. It may be hard to implement to meet all requirements.

Reply:
The script can download the external packages to the local cache so the user can use the content of this cache to upload the packages to any kind of target